quickwit-doc-mapping/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pub enum DocMappingType {
+    Dynamic,
+    Opentelemetry,
+    Static,
+}
+
+pub struct DocMappingMetadata {
```

Comment:
I think it could be 

```rust
enum DocMapping {
   Dynamic,  
   OpenTelemetry,
   Static(StaticDocMapping),
}
```
and we would just tweak serde to serialize it the way you are suggesting with your DocMappingMetadata structure.
(I forgot if this is the default or not)